modules/tf-aws-open-next-public-resources/main.tf

```diff
@@ -499,6 +514,145 @@ resource "aws_cloudfront_cache_policy" "cache_policy" {
   }
 }
 
+resource "aws_cloudfront_response_headers_policy" "response_headers" {
+  count = var.response_headers.deployment == "DETACH" || local.create_response_headers ? 1 : 0
+  name  = "${local.prefix}response-headers${local.suffix}"
+
+  dynamic "cors_config" {
+    for_each = try(var.response_headers.cors_config, null) != null ? [true] : []
+
+    content {
+      access_control_allow_credentials = try(var.response_headers.cors_config.access_control_allow_credentials, null)
+
+      access_control_allow_headers {
+        items = try(var.response_headers.cors_config.access_control_allow_headers, [])
+      }
+
+      access_control_allow_methods {
+        items = try(var.response_headers.cors_config.access_control_allow_methods, [])
+      }
+
+      access_control_allow_origins {
+        items = try(var.response_headers.cors_config.access_control_allow_origins, [])
+      }
+
+      dynamic "access_control_expose_headers" {
+        for_each = var.response_headers.cors_config.access_control_expose_headers
+
+        content {
+          items = var.response_headers.cors_config.access_control_expose_headers
+        }
+
+      }
```

Comment:
is this meant to create a block for each header in the list or is the for each supposed to check if the list is greater than zero then use the list?

Reply:
i.e. for_each = try(var.response_headers.cors_config.access_control_expose_headers, null) != null ? [true] : []

Reply:
Yes, this is meant to only create one block if the length is more than 0 and excluded it if the list is empty. I'll add a length check as it defaults to an empty list `access_control_expose_headers    = optional(list(string), [])`